src/sentry/notifications/notifications/activity/assigned.py

```diff
@@ -2,102 +2,74 @@
 
 from typing import Any, Mapping
 
-from sentry.models import Team, User
+from sentry.models import Activity, Organization, Team, User
 
 from .base import GroupActivityNotification
 
 
+def _get_user_option(assignee_id: int) -> User | None:
+    try:
+        return User.objects.get_from_cache(id=assignee_id)
+    except User.DoesNotExist:
+        return None
+
+
+def _get_team_option(assignee_id: int, organization: Organization) -> Team | None:
+    try:
+        return Team.objects.get(id=assignee_id, organization=organization)
```

Comment:
@mgaeta Nit: you could just do `Team.objects.filter().first` and avoid having to use the try/except